In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/netflix-rotten-tomatoes-metacritic-imdb/netflix-rotten-tomatoes-metacritic-imdb.csv


In [2]:
import pandas as pd
import numpy as np 
import matplotlib.pylab as plt
import seaborn as sns
plt.style.use('_mpl-gallery')
pd.set_option("display.max_columns", 300)

In [3]:
df = pd.read_csv('/kaggle/input/netflix-rotten-tomatoes-metacritic-imdb/netflix-rotten-tomatoes-metacritic-imdb.csv')
df.head()

Title                                   Genre  \
0     Lets Fight Ghost  Crime, Drama, Fantasy, Horror, Romance   
1  HOW TO BUILD A GIRL                                  Comedy   
2           Centigrade                         Drama, Thriller   
3                ANNE+                                   Drama   
4                Moxie                 Animation, Short, Drama   

                                                Tags         Languages  \
0  Comedy Programmes,Romantic TV Comedies,Horror ...  Swedish, Spanish   
1       Dramas,Comedies,Films Based on Books,British           English   
2                                          Thrillers           English   
3        TV Dramas,Romantic TV Dramas,Dutch TV Shows           Turkish   
4  Social Issue Dramas,Teen Movies,Dramas,Comedie...           English   

  Series or Movie  Hidden Gem Score  \
0          Series               4.3   
1           Movie               7.0   
2           Movie               6.4   
3          Series               7.7   
4           Movie               8.1   

                                Country Availability       Runtime  \
0                                           Thailand  < 30 minutes   
1                                             Canada      1-2 hour   
2                                             Canada      1-2 hour   
3                                Belgium,Netherlands  < 30 minutes   
4  Lithuania,Poland,France,Iceland,Italy,Spain,Gr...      1-2 hour   

          Director                      Writer  \
0  Tomas Alfredson       John Ajvide Lindqvist   
1    Coky Giedroyc               Caitlin Moran   
2    Brendan Walsh  Brendan Walsh, Daley Nixon   
3              NaN                         NaN   
4    Stephen Irwin                         NaN   

                                              Actors View Rating  IMDb Score  \
0  Kåre Hedebrant, Per Ragnar, Lina Leandersson, ...           R         7.9   
1  Paddy Considine, Cleo, Beanie Feldstein, Dónal...           R         5.8   
2                  Genesis Rodriguez, Vincent Piazza     Unrated         4.3   
3  Vahide Perçin, Gonca Vuslateri, Cansu Dere, Be...         NaN         6.5   
4                                       Ragga Gudrun         NaN         6.3   

   Rotten Tomatoes Score  Metacritic Score  Awards Received  \
0                   98.0              82.0             74.0   
1                   79.0              69.0              1.0   
2                    NaN              46.0              NaN   
3                    NaN               NaN              1.0   
4                    NaN               NaN              NaN   

   Awards Nominated For   Boxoffice Release Date Netflix Release Date  \
0                  57.0  $2,122,065  12 Dec 2008           2021-03-04   
1                   NaN     $70,632  08 May 2020           2021-03-04   
2                   NaN     $16,263  28 Aug 2020           2021-03-04   
3                   NaN         NaN  01 Oct 2016           2021-03-04   
4                   4.0         NaN  22 Sep 2011           2021-03-04   

                         Production House  \
0               Canal+, Sandrew Metronome   
1  Film 4, Monumental Pictures, Lionsgate   
2                                     NaN   
3                                     NaN   
4                                     NaN   

                             Netflix Link  \
0  https://www.netflix.com/watch/81415947   
1  https://www.netflix.com/watch/81041267   
2  https://www.netflix.com/watch/81305978   
3  https://www.netflix.com/watch/81336456   
4  https://www.netflix.com/watch/81078393   

                              IMDb Link  \
0  https://www.imdb.com/title/tt1139797   
1  https://www.imdb.com/title/tt4193072   
2  https://www.imdb.com/title/tt8945942   
3  https://www.imdb.com/title/tt6132758   
4  https://www.imdb.com/title/tt2023611   

                                             Summary  IMDb Votes  \
0  A med student with a supernatural gift tries t.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15480 entries, 0 to 15479
Data columns (total 29 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Title                  15480 non-null  object 
 1   Genre                  13770 non-null  object 
 2   Tags                   15413 non-null  object 
 3   Languages              13526 non-null  object 
 4   Series or Movie        15480 non-null  object 
 5   Hidden Gem Score       13379 non-null  float64
 6   Country Availability   15461 non-null  object 
 7   Runtime                15479 non-null  object 
 8   Director               10772 non-null  object 
 9   Writer                 11150 non-null  object 
 10  Actors                 13555 non-null  object 
 11  View Rating            8456 non-null   object 
 12  IMDb Score             13381 non-null  float64
 13  Rotten Tomatoes Score  6382 non-null   float64
 14  Metacritic Score       4336 non-null   float64
 15  Aw

**Data Cleaning**

handle missing values

In [5]:
df.shape

(15480, 29)

In [6]:
print("percentage of missing values")
df.isna().sum()/15480*100

percentage of missing values


Title                     0.000000
Genre                    11.046512
Tags                      0.432817
Languages                12.622739
Series or Movie           0.000000
Hidden Gem Score         13.572351
Country Availability      0.122739
Runtime                   0.006460
Director                 30.413437
Writer                   27.971576
Actors                   12.435401
View Rating              45.374677
IMDb Score               13.559432
Rotten Tomatoes Score    58.772610
Metacritic Score         71.989664
Awards Received          60.755814
Awards Nominated For     50.510336
Boxoffice                74.114987
Release Date             13.611111
Netflix Release Date      0.000000
Production House         66.737726
Netflix Link              0.000000
IMDb Link                14.877261
Summary                   0.058140
IMDb Votes               13.572351
Image                     0.000000
Poster                   23.501292
TMDb Trailer             53.527132
Trailer Site        

In [7]:
#remove columns that have too high a percentage of missing values
df = df.drop(columns = ['Boxoffice', 'Metacritic Score', 'TMDb Trailer', 'Trailer Site', 'Poster'], axis=1)

In [8]:
#percentage of missing values in columns that have object data type
object_kolumns = df.select_dtypes(include=['object']).columns
df_object=df[object_kolumns]
df_object.isna().sum()

Title                       0
Genre                    1710
Tags                       67
Languages                1954
Series or Movie             0
Country Availability       19
Runtime                     1
Director                 4708
Writer                   4330
Actors                   1925
View Rating              7024
Release Date             2107
Netflix Release Date        0
Production House        10331
Netflix Link                0
IMDb Link                2303
Summary                     9
Image                       0
dtype: int64

In [9]:
#resolve missing values in columns that have object data type
columns_to_fill = ['Genre', 'Tags', 'Languages', 'Country Availability', 'Director', 'Writer', 'Actors', 'Release Date', 'Production House', 'IMDb Link', 'Summary']

# Fill the missing value in the column with 'unknown'
df[columns_to_fill] = df[columns_to_fill].fillna('unknown')

# Fill in the missing values in the 'View Rating' and 'Runtime' columns
df['View Rating'] = df['View Rating'].fillna('Unrated')
df['Runtime'] = df['Runtime'].fillna('unknown')



In [10]:
#percentage of missing values in columns that have float data type
float_columns = df.select_dtypes(include=['float64']).columns
float_object=df[float_columns]
float_object.isna().sum()/15480*100

Hidden Gem Score         13.572351
IMDb Score               13.559432
Rotten Tomatoes Score    58.772610
Awards Received          60.755814
Awards Nominated For     50.510336
IMDb Votes               13.572351
dtype: float64

In [11]:
float_object.describe()

Hidden Gem Score    IMDb Score  Rotten Tomatoes Score  Awards Received  \
count      13379.000000  13381.000000            6382.000000      6075.000000   
mean           5.937551      6.496054              59.523034         8.764444   
std            2.250202      1.146910              26.999173        18.311171   
min            0.600000      1.000000               0.000000         1.000000   
25%            3.800000      5.800000              38.000000         1.000000   
50%            6.800000      6.600000              64.000000         3.000000   
75%            7.900000      7.300000              83.000000         8.000000   
max            9.800000      9.700000             100.000000       300.000000   

       Awards Nominated For    IMDb Votes  
count           7661.000000  1.337900e+04  
mean              13.983161  4.272841e+04  
std               29.821052  1.257012e+05  
min                1.000000  5.000000e+00  
25%                2.000000  4.035000e+02  
50%                5.000000  2.322000e+03  
75%               12.000000  2.089050e+04  
max              386.000000  2.354197e+06

In [12]:
float_object.columns

Index(['Hidden Gem Score', 'IMDb Score', 'Rotten Tomatoes Score',
       'Awards Received', 'Awards Nominated For', 'IMDb Votes'],
      dtype='object')

In [13]:
float_columns = df.select_dtypes(include=['float64']).columns
float_object = df[float_columns]

# Menghitung rata-rata dan mengisi nilai yang hilang dengan rata-rata
min_float = float_object.min().round()
df[float_columns] = df[float_columns].fillna(min_float)


In [14]:
#percentage of missing values in columns that have float data type
float_columns = df.select_dtypes(include=['float64']).columns
float_object=df[float_columns]
float_object.isna().sum()/15480*100

Hidden Gem Score         0.0
IMDb Score               0.0
Rotten Tomatoes Score    0.0
Awards Received          0.0
Awards Nominated For     0.0
IMDb Votes               0.0
dtype: float64

Too many missing values have risks if you delete them immediately, I decided to fill in some missing columns to maximize the performance of the data for further analysis.

formating data

In [15]:
df['Netflix Release Date'] = pd.to_datetime(df['Netflix Release Date'])

In [16]:
df.head()

Title                                   Genre  \
0     Lets Fight Ghost  Crime, Drama, Fantasy, Horror, Romance   
1  HOW TO BUILD A GIRL                                  Comedy   
2           Centigrade                         Drama, Thriller   
3                ANNE+                                   Drama   
4                Moxie                 Animation, Short, Drama   

                                                Tags         Languages  \
0  Comedy Programmes,Romantic TV Comedies,Horror ...  Swedish, Spanish   
1       Dramas,Comedies,Films Based on Books,British           English   
2                                          Thrillers           English   
3        TV Dramas,Romantic TV Dramas,Dutch TV Shows           Turkish   
4  Social Issue Dramas,Teen Movies,Dramas,Comedie...           English   

  Series or Movie  Hidden Gem Score  \
0          Series               4.3   
1           Movie               7.0   
2           Movie               6.4   
3          Series               7.7   
4           Movie               8.1   

                                Country Availability       Runtime  \
0                                           Thailand  < 30 minutes   
1                                             Canada      1-2 hour   
2                                             Canada      1-2 hour   
3                                Belgium,Netherlands  < 30 minutes   
4  Lithuania,Poland,France,Iceland,Italy,Spain,Gr...      1-2 hour   

          Director                      Writer  \
0  Tomas Alfredson       John Ajvide Lindqvist   
1    Coky Giedroyc               Caitlin Moran   
2    Brendan Walsh  Brendan Walsh, Daley Nixon   
3          unknown                     unknown   
4    Stephen Irwin                     unknown   

                                              Actors View Rating  IMDb Score  \
0  Kåre Hedebrant, Per Ragnar, Lina Leandersson, ...           R         7.9   
1  Paddy Considine, Cleo, Beanie Feldstein, Dónal...           R         5.8   
2                  Genesis Rodriguez, Vincent Piazza     Unrated         4.3   
3  Vahide Perçin, Gonca Vuslateri, Cansu Dere, Be...     Unrated         6.5   
4                                       Ragga Gudrun     Unrated         6.3   

   Rotten Tomatoes Score  Awards Received  Awards Nominated For Release Date  \
0                   98.0             74.0                  57.0  12 Dec 2008   
1                   79.0              1.0                   1.0  08 May 2020   
2                    0.0              1.0                   1.0  28 Aug 2020   
3                    0.0              1.0                   1.0  01 Oct 2016   
4                    0.0              1.0                   4.0  22 Sep 2011   

  Netflix Release Date                        Production House  \
0           2021-03-04               Canal+, Sandrew Metronome   
1           2021-03-04  Film 4, Monumental Pictures, Lionsgate   
2           2021-03-04                                 unknown   
3           2021-03-04                                 unknown   
4           2021-03-04                                 unknown   

                             Netflix Link  \
0  https://www.netflix.com/watch/81415947   
1  https://www.netflix.com/watch/81041267   
2  https://www.netflix.com/watch/81305978   
3  https://www.netflix.com/watch/81336456   
4  https://www.netflix.com/watch/81078393   

                              IMDb Link  \
0  https://www.imdb.com/title/tt1139797   
1  https://www.imdb.com/title/tt4193072   
2  https://www.imdb.com/title/tt8945942   
3  https://www.imdb.com/title/tt6132758   
4  https://www.imdb.com/title/tt2023611   

                                             Summary  IMDb Votes  \
0  A med student with a supernatural gift tries t...    205926.0   
1  When nerdy Johanna moves to London, things get...      2838.0   
2  Trapped in a frozen car during a blizzard, a p...      1720.0   
3  Upon moving into a new place, a 20-something r...     

**conclusion**

there are too many missing values, some even reach the percentage of 74% missing values from the whole data, I choose to delete the column that has too many missing values and fill the others using the lowest value in the data.